src/theory/strings/infer_proof_cons.cpp

```diff
@@ -0,0 +1,1043 @@
+        {
+          Trace("strings-ipc-red") << "...success!" << std::endl;
+          useBuffer = true;
+        }
+        else
+        {
+          Trace("strings-ipc-red") << "...failed to reduce" << std::endl;
+        }
+      }
+    }
+    break;
+    // ========================== code injectivity
+    case Inference::CODE_INJ:
+    {
+      ps.d_rule = PfRule::STRING_CODE_INJ;
+      Assert(conc.getKind() == OR && conc.getNumChildren() == 3
+             && conc[2].getKind() == EQUAL);
+      ps.d_args.push_back(conc[2][0]);
+      ps.d_args.push_back(conc[2][1]);
+    }
+    break;
+    // ========================== unit injectivity
+    case Inference::UNIT_INJ: { ps.d_rule = PfRule::STRING_SEQ_UNIT_INJ;
+    }
+    break;
+    // ========================== prefix conflict
+    case Inference::PREFIX_CONFLICT:
+    {
+      Trace("strings-ipc-prefix") << "Prefix conflict..." << std::endl;
+      std::vector<Node> eqs;
+      for (const Node e : ps.d_children)
+      {
+        Kind ek = e.getKind();
+        if (ek == EQUAL)
+        {
+          Trace("strings-ipc-prefix") << "- equality : " << e << std::endl;
+          eqs.push_back(e);
+        }
+        else if (ek == STRING_IN_REGEXP)
+        {
+          // unfold it and extract the equality
+          std::vector<Node> children;
+          children.push_back(e);
+          std::vector<Node> args;
+          Node eunf = psb.tryStep(PfRule::RE_UNFOLD_POS, children, args);
+          Trace("strings-ipc-prefix")
+              << "--- " << e << " unfolds to " << eunf << std::endl;
+          if (eunf.isNull())
+          {
+            continue;
+          }
+          else if (eunf.getKind() == AND)
+          {
+            // equality is the last conjunct
+            std::vector<Node> childrenAE;
+            childrenAE.push_back(eunf);
+            std::vector<Node> argsAE;
+            argsAE.push_back(nm->mkConst(Rational(eunf.getNumChildren() - 1)));
+            Node eunfAE = psb.tryStep(PfRule::AND_ELIM, childrenAE, argsAE);
+            Trace("strings-ipc-prefix")
+                << "--- and elim to " << eunfAE << std::endl;
+            if (eunfAE.isNull() || eunfAE.getKind() != EQUAL)
+            {
+              Assert(false) << "Unexpected unfolded premise " << eunf << " for " << infer;
+              continue;
+            }
+            Trace("strings-ipc-prefix")
+                << "- equality : " << eunfAE << std::endl;
+            eqs.push_back(eunfAE);
+          }
+          else if (eunf.getKind() == EQUAL)
+          {
+            Trace("strings-ipc-prefix") << "- equality : " << eunf << std::endl;
+            eqs.push_back(eunf);
+          }
+        }
+        else
+        {
+          // not sure how to use this assumption
+          Assert(false) << "Unexpected premise " << e << " for " << infer;
+        }
+      }
+      if (eqs.empty())
+      {
+        break;
+      }
+      // connect via transitivity?
```

Comment:
```suggestion
      // connect via transitivity
```

Or why is there a question mark?